packages/canary-web-components/src/components/ic-data-table/ic-data-table.tsx

```diff
@@ -795,23 +801,27 @@ export class DataTable {
             }
           ></ic-loading-indicator>
         )}
-        {showPagination && (
+        {(showPagination || isSlotUsed(this.el, "pagination-bar")) && (
```

Comment:
It would be good to check if there are any issues with not having any slot content initially, but then adding it later, as there have been issues with other components with this approach. A bit of an edge case, but I'm thinking of maybe you have an empty table initially & only put content in the bar slot once data has loaded